src/Functional/CompareObjectHashOn.php

```diff
@@ -13,9 +13,14 @@
 /**
  * Returns a comparison function that can be used with e.g. `usort()`
  *
- * @param callable $comparison A function that compares the two values. Pick e.g. strcmp() or strnatcasecmp()
- * @param callable $keyFunction A function that takes an argument and returns the value that should be compared
- * @return callable
+ * @template V
+ * @template R of int
```

Comment:
Could be more specific
```suggestion
 * @template R of -1|0|1
```

Reply:
I stuck to `int` since before `php 8.2` it is still the case for functions like [strnatcmp](https://www.php.net/manual/en/function.strnatcmp.php#refsect1-function.strnatcmp-changelog)
But I can force to `-1|0|1` if you really think we should

Reply:
Nah, it’s fine then